src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs

```diff
@@ -11,6 +11,34 @@ internal static partial class Interop
     internal static partial class Advapi32
     {
         [DllImport(Interop.Libraries.Advapi32, EntryPoint = "LsaOpenPolicy", SetLastError = true, CharSet = CharSet.Unicode)]
-        internal static extern uint LsaOpenPolicy(string systemName, ref LSA_OBJECT_ATTRIBUTES attributes, int accessMask, out SafeLsaPolicyHandle handle);
+        private static extern uint LsaOpenPolicy(
+            ref SspiCli.LSA_UNICODE_STRING SystemName,
```

Comment:
What does the type have to do with SSPI?  It seems just generically Advapi32 to me.

Reply:
Strictly speaking it's actually NT kernel type (UNICODE_STRING) so NTDLL feels somewhat appropriate too.

Reply:
We already have a `UNICODE_STRING` type I think

Reply:
We should consider using the existing `UNICODE_STRING` and get rid of the other. It is actually defined that way internally:

``` C
NTSTATUS
LsaOpenPolicy(
    IN PUNICODE_STRING SystemName OPTIONAL,
    IN POBJECT_ATTRIBUTES ObjectAttributes,
    IN ACCESS_MASK DesiredAccess,
    IN OUT PLSA_HANDLE PolicyHandle
    )
```

There really is only one `UNICODE_STRING` in NT.

Reply:
We're doing a `Mashal.Sizeof` for the object attributes in the calling code. We may just want to follow suit with what I did here: https://github.com/dotnet/corefx/blob/2f7844a8df4fef427df1a2bc77e9f34211a720a1/src/Common/src/Interop/Windows/NtDll/Interop.NtCreateFile.cs#L49

And use the `OBJECT_ATTRIBUTES`, killing the LSA version.